### PR TITLE
fix(info): mask username on Windows

### LINKED ISSUE #2731: aqua info: user name isn't masked on Windows
### Discussed in https://github.com/orgs/aquaproj/discussions/2511

<div type='discussions-op-text'>

<sup>Originally posted by **suzuki-shunsuke** November 23, 2023</sup>
### aqua info

```console
$ aqua info
{
"version": "2.16.0",
"commit_hash": "70f84d66ba9af98008360fd7b846094f020b4c6f",
"os": "windows",
"arch": "amd64",
"pwd": "C:\Users\devnu",
"root_dir": "C:\Users\devnu\AppData\Local\aquaproj-aqua",
"env": {},
"config_files": []
}
```

### Overview

`aqua info` should mask user name.
But user name isn't masked on Windows.

### How to reproduce

Run `aqua info` on Windows.
I confirmed the issue with Git Bash and Powershell.

```console
$ aqua info
```

### Debug output

_No response_

### Expected behaviour

`aqua info` should mask user name.

### Actual behaviour

`aqua info` doesn't mask user name on Windows.

### Important Factoids

_No response_

### Reference

_No response_</div>

### PR DESCRIPTION
**Pull request by sapphi-red**
## Check List

- [x] Read [CONTRIBUTING.md](https://github.com/aquaproj/aqua/blob/main/CONTRIBUTING.md)
  - [x] Read [OSS Contribution Guide](https://github.com/suzuki-shunsuke/oss-contribution-guide/blob/main/README.md)
- [x] [Write a GitHub Issue before creating a Pull Request](https://github.com/suzuki-shunsuke/oss-contribution-guide/blob/main/README.md#create-an-issue-before-creating-a-pull-request)
  - Link to the issue: fixes #2731
- [x] [All commits are signed](https://github.com/suzuki-shunsuke/oss-contribution-guide/blob/main/docs/commit-signing.md)
  - This repository enables `Require signed commits`, so all commits must be signed
- [x] [Avoid force push](https://github.com/suzuki-shunsuke/oss-contribution-guide?tab=readme-ov-file#dont-do-force-pushes-after-opening-pull-requests)
- [x] Do only one thing in one Pull Request

On windows, [`user.Current().Username` returns a string with domain and user concatenated](https://github.com/golang/go/blob/1667dbd7be0da5e75a25f14c339c859ed2190b43/src/os/user/lookup_windows.go#L180-L188). This seems to be called as [down-level logon name](https://learn.microsoft.com/en-us/windows/win32/secauthn/user-name-formats#down-level-logon-name).
This PR makes the domain part to be trimmed, so that the username is masked when running the info command.

### CODE REVIEW

File: pkg/controller/info/info.go [hunk 1]
```diff
@@ -50,12 +50,25 @@ func maskUser(s, username string) string {
...
 
-func (c *Controller) Info(_ context.Context, _ *logrus.Entry, param *config.Param) error { //nolint:funlen
+func getCurrentUserName() (string, error) {
 	currentUser, err := user.Current()
 	if err != nil {
-		return fmt.Errorf("get a current user: %w", err)
+		return "", fmt.Errorf("get a current user: %w", err)
 	}
 	userName := currentUser.Username
+	if runtime.GOOS == "windows" {
```
Comment (sapphi-red) [inline pkg/controller/info/info.go:59]: I used a runtime detection instead of compile time as I didn't think the complexity is worth the benefit.

**Comment by suzuki-shunsuke** [comment]
Thank you for your contribution!

File: pkg/controller/info/info.go [hunk 2]
```diff
@@ -50,12 +50,25 @@ func maskUser(s, username string) string {
...
+		// On Windows, the user name is in the form of "domain\user".
+		const domainUserPartCount = 2
+		userName = strings.SplitN(userName, "\\", domainUserPartCount)[domainUserPartCount-1]
```
Comment (suzuki-shunsuke) [inline pkg/controller/info/info.go:62]: I think strings.Cut is better than strings.SplitN.

https://pkg.go.dev/strings#Cut

```suggestion
		domain, un, ok := strings.Cut(userName, "\\")
		if ok {
			userName = un
		} else {
			userName = domain
		}
```

Comment (sapphi-red) [reply pkg/controller/info/info.go:62]: Ah, I forgot that function exists.

**Comment by suzuki-shunsuke** [comment]
v2.28.1 is out 🎉
https://github.com/aquaproj/aqua/releases/tag/v2.28.1

**Pull request merged**